solutions/src/main/java/ru/sberbank/school/task13/cache/CacheProxy.java

```diff
@@ -0,0 +1,334 @@
+            } else {
+                result = method.invoke(service, args);
+                writeToCacheFile(file, result);
+            }
+
+            return result;
+        }
+
+        private Object findInCache(String toFind) {
+            Object result = null;
+
+            if (cacheMap.containsKey(toFind)) {
+                result = cacheMap.get(toFind);
+            } else {
+                for (String keyFromCacheMap : cacheMap.keySet()) {
+                    if (keyFromCacheMap.contains(toFind)) {
+                        result = cacheMap.get(keyFromCacheMap);
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        private File findInDirectory(File toFind) {
+            String fullNameToFind = isZip(toFind.getAbsolutePath()) ? (toFind.getAbsolutePath() + ".zip") : toFind.getAbsolutePath();
+            File result = null;
+            File toCheck = new File(fullNameToFind);
+
+            if (toCheck.exists() && ! toCheck.isDirectory()) {
+                result = new File(fullNameToFind);
+            }
+            return result;
+        }
+
+        private Object loadFromCacheFile(File fileToLoad) {
+            Object result = null;
+
+            if (isZip(fileToLoad.getName())) {
+                result = readZipFile(fileToLoad);
+            } else {
+                result = readFromFile(fileToLoad);
+            }
+
+            return result;
+        }
+
+        private Object readFromFile(File file) {
+            Object result = null;
+
+            try (FileInputStream fin = new FileInputStream(file);
+                 ObjectInputStream oin = new ObjectInputStream(fin)) {
+
+                result = oin.readObject();
+
+            } catch (ClassNotFoundException | IOException e) {
+                throw new CacheProxyException("Error occurred while loading file: " + file.getName(), e);
+            }
+            return result;
+        }
+
+        private Object readZipFile(File file) {
+            Object result = null;
+            try {
+                File zipFileToGet = new File(file.getAbsolutePath());
+
+                if (zipFileToGet.exists()) {
+                    try (ZipFile zipFile = new ZipFile(file.getAbsolutePath())) {
+                        ZipEntry zipEntry = zipFile.getEntry(file.getName().replace(".zip", ""));
+
+                        try (InputStream in = zipFile.getInputStream(zipEntry)) {
+                            byte[] bytes = getBytes(in);
+
+                            try (ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(bytes))) {
+                                result = ois.readObject();
+                            }
+                        }
+                    }
+                }
+            } catch (IOException | ClassNotFoundException e) {
+                throw new CacheProxyException("Error occurred while loading file: " + file.getName() +
```

Comment:
&#39;+&#39; should be on a new line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1657330847>*